.github/actions/asdf-install-tooling/action.yml

```diff
@@ -0,0 +1,124 @@
+---
+name: Install asdf and tools
+description: |
+    Install asdf and all tools listed in the .tool-versions file.
+    This action will:
+      - Check if the tools are present in the cache, otherwise it will:
+      - Download and verify the asdf binary
+      - Extract and install it in ~/.local/bin
+      - Automatically install all plugins listed in .tool-versions
+      - Install the required tool versions using asdf
+      - Cache the installed tools
+
+
+inputs:
+    version:
+        description: asdf version
+        # renovate: datasource=github-tags depName=asdf-vm/asdf
+        default: v0.16.0
+    os:
+        description: Target OS (linux or darwin)
+        default: linux
+    arch:
+        description: Target architecture (amd64, arm64, etc.)
+        default: amd64
+    cache:
+        description: Use cache for tools installed
+        default: 'true'
+
+runs:
+    using: composite
+    steps:
+        - name: Compute cache key
+          id: cache-key
+          shell: bash
+          run: |
+              WEEK_NUMBER="$(date +%V)"
```

Comment:
why do we need to invalidate the cache on a weekly basis?

Reply:
as mentioned in a pm, would go with arch + if possible runs-on name instead of week.
See my [reasoning](https://github.com/camunda/infraex-common-config/pull/174#issuecomment-26101233890) on why the weekly invalidation is not a permanent fix.